social_protection/validation.py

```diff
@@ -7,12 +11,59 @@ class BenefitPlanValidation(BaseModelValidation):
 
     @classmethod
     def validate_create(cls, user, **data):
+        incoming_code = data.get('code')
+        code_error = check_bf_unique_code(incoming_code)
+        if code_error:
+            raise ValidationError(code_error[0]['message'])
+        incoming_name = data.get('name')
+        name_error = check_bf_unique_name(incoming_name)
+        if name_error:
+            raise ValidationError(name_error[0]['message'])
+        incoming_schema = data.get('beneficiary_data_schema')
+        schema_error = is_valid_json_schema(incoming_schema)
+        if schema_error:
+            raise ValidationError(schema_error[0]['message'])
```

Comment:
I think it'd be better to go through all checks that are possible and then return more complex validation error. For instance if you'd send bad request with duplicated name and code, then instead of sending one request and fixing it, you'd have to send 2 of them. It's not an issue with this amount of data but when entity will get more complex and have 30 attributes then debugging would be annoying. 
You can change it to something like: 
```suggestion
        errors = [
        *check_bf_unique_code(data.get('code'), 
        *check_bf_unique_name(data.get('name')),
        *is_valid_json_schema( data.get('beneficiary_data_schema'))
        ]
        if errors:
            raise ValidationError(errors)
```

Reply:
done